### PR TITLE
Fix slicing for padding + cache

### PR DESCRIPTION
**Pull request by dakinggg**
Fixes a bug in the position id adjustment when using kv caching and padding together. Previously, the slicing happened to the `attention_mask == 0` part, but should have happened outside of the `cumsum` part. This only had an effect when doing both kv caching and padding together, so definitely should not affect training, only generation.

### CODE REVIEW

**Review by vchiley — APPROVED** [review]
lgtm
